apps/openmw/mwphysics/physicssystem.cpp

```diff
@@ -326,7 +327,30 @@ namespace MWPhysics
             if (movement.z() > 0 && ptr.getClass().getCreatureStats(ptr).isDead() && position.z() < swimlevel)
                 velocity = osg::Vec3f(0,0,1) * 25;
 
-            ptr.getClass().getMovementSettings(ptr).mPosition[2] = 0;
+            if (ptr.getClass().getMovementSettings(ptr).mPosition[2])
```

Comment:
Can there be a situation when the `mPosition[2]` is negative?

Reply:
Currently not, it's just a "vertical movement attempt" flag expressed in a number.